aten/src/ATen/core/function_schema.h

```diff
@@ -78,6 +95,38 @@ struct Argument {
     return Argument(name_, new_type, N_, default_value_, kwarg_only_, alias_info_);
   }
 
+  // this function check whether this Argument is backward compatible with
+  // the old_argument. we intent to be conservative, if necessary, we may
+  // relax the checks in future. we consider the following the situations are
+  // backward compatible:
+  //   1) two arguments are equal
+  //   2) old_argument's type is T, and this Argument's type is Optional[T]
+  //   3) old_argument has no default value, and this Argument provides default
+  //      value
+  bool isBackwardCompatibleWith(const Argument& old_argument) const {
+    if (*this == old_argument) {
+      return true;
+    }
+    if (!(name() == old_argument.name()
+        && kwarg_only() == old_argument.kwarg_only()
+        && N() == old_argument.N()
+        && alias_info() == old_argument.alias_info())) {
+      return false;
+    }
+    if (!(*type() == *old_argument.type() ||
+          (type()->isSubclass(TypeKind::OptionalType) &&
```

Comment:
Please use `type()->cast<OptionalType>()` to test for it being optional, `isSubclass` is for internal use handling actual C++ subclassing.

Reply:
I just saw `isSubtypeOfExt`, and used it.